docs/samples/multimodelserving/benchmark/BENCHMARK.md

```diff
@@ -0,0 +1,152 @@
+# Benchmark of Multi-Model Serving
+
+The following stress and performance tests have been executed to support that Multi-Model Serving is a solution to the model deployment scalability problem as discussed in the [MMS Guide](../../../MULTIMODELSERVING_GUIDE.md).
+
+## Stress Test
+To determine the maximum amount of models that can be deployed
+- Environment Setup:
+    - kfserving: v0.5.1
+    - knative: v0.17.0
+    - istio: v1.7.x
+- Cluster limitation:
+    - Maximum 2048 IP Addresses
+    - Maximum 0.55 TiB memory per worker node
+    - Maximum ~2000 kubernetes services resources
+        - Limited by istio ingress gateway
+    - Maximum 660 pods on worker nodes
+        - Maximum 110 pods per worker node
+        - 6 total worker nodes
+    - ETCD size of 65 MiB
+- Inference Service:
+    - apiVersion: `serving.kubeflow.org/v1beta1`
+    - concurrency: 1
+    - minReplicas: 1
+    - Triton model server
+    - Queue_proxy: CPU 1, Memory 500Mi
+    - Traditional Model Serving Predictor Spec: CPU 100m, Memory 100Mi
+    - Multi-Model Serving Predictor Spec: CPU 10, Memory 10Gi
+- Model:
+    - [Simple string](https://github.com/triton-inference-server/server/tree/b5a5541d3252a9fb235eeea9c156eb0d2f8ab883/docs/examples/model_repository/simple_string)
+    - Size of 700B
+    - Tensorflow framework
+
+### Baseline:
+- Test Summary:
+    - The traditional model serving was able to deploy `352 InferenceServices`
+    - Multi-model serving was able to deploy `~10K models using 6 InferenceService` which equates to `2800%` more models loaded
+    - Overall, Multi-Model Serving utilises less CPU and memory while deploying more
+        - The traditional model serving used a total of `387 CPU` and `211 GB`
+        - Multi-model serving used a total of `66 CPU` and `63 GB`
+        - Comparing the queue-proxy usage
+            - The traditional model serving queue-proxies used `352 CPU` and `176 GB`
+            - Multi-model serving queue-proxies used `6 CPU` and `3 GB`
+    - Adding/removing models in multi-model serving causes new revisions to be generated without the previous being removed which can be harmful to etcd when deploying thousands of models
```

Comment:
```suggestion
    - Adding/removing models in multi-model serving causes new configmap versions to be generated without ETCD getting the chance to remove the previous versions. This can exhaust the memory of ETCD in case of deploying thousands of models
```